localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/gateway_exception.py

```diff
@@ -0,0 +1,68 @@
+import json
+import logging
+
+from rolo import Response
+
+from localstack.services.apigateway.next_gen.execute_api.api import (
+    RestApiGatewayExceptionHandler,
+    RestApiGatewayHandlerChain,
+)
+from localstack.services.apigateway.next_gen.execute_api.context import RestApiInvocationContext
+from localstack.services.apigateway.next_gen.execute_api.gateway_response import (
+    BaseGatewayException,
+    get_gateway_response_or_default,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class GatewayExceptionHandler(RestApiGatewayExceptionHandler):
+    """
+    Exception handler that serializes the Gateway Exceptions into Gateway Responses
+    """
+
+    def __call__(
+        self,
+        chain: RestApiGatewayHandlerChain,
+        exception: Exception,
+        context: RestApiInvocationContext,
+        response: Response,
+    ):
+        if not isinstance(exception, BaseGatewayException):
+            LOG.warning(
+                "Non Gateway Exception raised: %s",
+                exception,
+                exc_info=LOG.isEnabledFor(logging.DEBUG),
+            )
+            response.update_from(
+                Response(response=f"Error in apigateway invocation: {exception}", status="500")
```

Comment:
I think this is good, it also makes client fail trying to decode JSON, really indicating that something is wrong. Nice!